oracle/oracle.go

```diff
@@ -0,0 +1,120 @@
+package oracle
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/republicprotocol/republic-go/crypto"
+	"github.com/republicprotocol/republic-go/dispatch"
+	"github.com/republicprotocol/republic-go/identity"
+	"github.com/republicprotocol/republic-go/logger"
+	"github.com/republicprotocol/republic-go/swarm"
+)
+
+// A Client exposes methods for invoking RPCs on a remote server.
+type Client interface {
+	// UpdateMidpoint is used to send updated midpoint information to a given
+	// multiAddress.
+	UpdateMidpoint(ctx context.Context, to identity.MultiAddress, midpointPrice MidpointPrice) error
+
+	// MultiAddress is used when finding random nodes to send information to.
```

Comment:
The comment "is used" is vague. How is it used? Is this method returning a random `MultiAddress` whenever it is called? Interface comments need to be sufficiently in-depth that someone can come and build an implementation for the interface.